pyuvsim/source.py

```diff
@@ -157,8 +166,8 @@ def pos_lmn(self, time, telescope_location):
             (l, m, n) direction cosine values
         """
         # calculate direction cosines of source at current time and array location
-        if self.alt_az is None:
-            self.alt_az_calc(time, telescope_location)
+        # Will only do the calculation if time has changed
+        self.alt_az_calc(time, telescope_location)
```

Comment:
I don't think this comment is true currently. I think for it to be true you need to call your new function `get_alt_az`.

Reply:
You're right... assuming this doesn't break things completely, it might mean that we'd actually get a further speedup, since it wasn't actually reusing previously-calculated alt/az values.